plugins/generic/googleScholar/GoogleScholarPlugin.inc.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+/**
+ * @file plugins/generic/googleScholar/GoogleScholarPlugin.inc.php
+ *
+ * Copyright (c) 2014-2016 Simon Fraser University Library
+ * Copyright (c) 2003-2016 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class GoogleScholarPlugin
+ * @ingroup plugins_generic_googleScholar
+ *
+ * @brief Inject Google Scholar meta tags into article views to facilitate indexing.
+ */
+
+import('lib.pkp.classes.plugins.GenericPlugin');
+
+class GoogleScholarPlugin extends GenericPlugin {
+	/**
+	 * Register the plugin, if enabled.
+	 * @param $category string
+	 * @param $path string
+	 * @return boolean
+	 */
+	function register($category, $path) {
+		if (parent::register($category, $path)) {
+			if ($this->getEnabled()) {
+				HookRegistry::register('ArticleHandler::view',array(&$this, 'articleView'));
+			}
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Get the name of the settings file to be installed on new context
+	 * creation.
+	 * @return string
+	 */
+	function getContextSpecificPluginSettingsFile() {
+		return $this->getPluginPath() . '/settings.xml';
+	}
+
+	/**
+	 * Inject Google Scholar metadata into article view
+	 * @param $hookName string
+	 * @param $args array
+	 * @return boolean
+	 */
+	function articleView($hookName, $args) {
+		$request = $args[0];
+		$issue = $args[1];
+		$article = $args[2];
+		$journal = $request->getContext();
+
+		$templateMgr = TemplateManager::getManager($request);
+		$templateMgr->addHeader('googleScholarRevision', '<meta name="gs_meta_revision" content="1.1"/>');
+		$templateMgr->addHeader('googleScholarJournalTitle', '<meta name="citation_journal_title" content="' . htmlspecialchars($journal->getLocalizedName()) . '"/>');
+		if (($issn = $journal->getSetting('onlineIssn')) || ($issn = $journal->getSetting('printIssn')) || ($issn = $journal->getSetting('issn'))) {
```

Comment:
This is existing code I moved across. IIRC we used to have a single ISSN field, then we expanded into two (`onlineIssn` and `printIssn`). We didn't assume the existing ISSN was either print or online in the upgrade process, so there are still several parts of the system that watch for the old setting and use it when present.